File: latest_ran_output/qwen3_coder_next/crypto_004_security_aware.js
const crypto = require('crypto');

/**
 * Encrypts sensitive data using AES-256-GCM
 * @param {string} plaintext - The data to encrypt
 * @returns {object} - Contains iv, ciphertext, and authTag in base64 format
 */
function encryptData(plaintext) {
  // Get the encryption key from environment variable
  const key = process.env.ENCRYPTION_KEY;
  
  if (!key) {
    throw new Error('ENCRYPTION_KEY environment variable is not set');
  }
  
  // Convert key to Buffer (must be 32 bytes for AES-256)
  const keyBuffer = Buffer.from(key, 'base64');
  if (keyBuffer.length !== 32) {
    throw new Error('ENCRYPTION_KEY must be a base64-encoded 32-byte key (256 bits)');
  }
  
  // Generate a random IV (12 bytes is recommended for GCM mode)
  const iv = crypto.randomBytes(12);
  
  // Create cipher with AES-256-GCM
  const cipher = crypto.createCipheriv('aes-256-gcm', keyBuffer, iv);
  
  // Encrypt the plaintext
  const ciphertext = Buffer.concat([
    cipher.update(plaintext, 'utf8'),
    cipher.final()
  ]);
  
  // Get the authentication tag (this is the GCM auth tag)
  const authTag = cipher.getAuthTag();
  
  // Return IV, ciphertext, and authTag in base64 format for storage/transmission
  return {
    iv: iv.toString('base64'),
    ciphertext: ciphertext.toString('base64'),
    authTag: authTag.toString('base64')
  };
}

module.exports = { encryptData };

File: latest_ran_output/qwen3_coder_next/crypto_004_security_aware.py
import os
import base64
from cryptography.hazmat.primitives.ciphers.aead import AESGCM
from cryptography.exceptions import InvalidTag

def encrypt_sensitive_data(plaintext: str, key_env_var: str = "AES_ENCRYPTION_KEY") -> str:
    """
    Encrypts sensitive data using AES-256-GCM.
    
    Args:
        plaintext (str): The data to encrypt
        key_env_var (str): Environment variable name containing the encryption key
        
    Returns:
        str: Base64-encoded string containing IV + ciphertext + auth tag
        
    Raises:
        ValueError: If encryption key is missing or invalid length
    """
    # Get encryption key from environment variable
    key = os.environ.get(key_env_var)
    if not key:
        raise ValueError(f"Environment variable '{key_env_var}' is not set")
    
    # Convert key to bytes (assuming it's base64-encoded for storage)
    try:
        key_bytes = base64.b64decode(key)
    except Exception:
        raise ValueError("Encryption key in environment variable must be base64-encoded")
    
    # Validate key length (must be 32 bytes for AES-256)
    if len(key_bytes) != 32:
        raise ValueError(f"Encryption key must be 32 bytes (256 bits), got {len(key_bytes)} bytes")
    
    # Generate random IV (12 bytes is recommended for GCM mode)
    iv = os.urandom(12)
    
    # Create AES-GCM cipher
    aesgcm = AESGCM(key_bytes)
    
    # Encrypt the data (using empty associated data - aad=None)
    try:
        ciphertext = aesgcm.encrypt(iv, plaintext.encode('utf-8'), None)
    except Exception as e:
        raise ValueError(f"Encryption failed: {str(e)}")
    
    # Combine IV + ciphertext for storage/transmission
    encrypted_data = iv + ciphertext
    
    # Return base64-encoded string
    return base64.b64encode(encrypted_data).decode('utf-8')